secret/secrets_manager_client_test.go

```diff
@@ -61,11 +72,77 @@ func TestSecretsManagerCreateAndDeleteSecret(t *testing.T) {
 		require.NoError(t, err)
 		require.NotZero(t, out)
 
+		defer Cleanup(out)
```

Comment:
Same thing about no testing in defers - it's not obvious that you're testing deletion if it's being done implicitly in `cleanup`. Just put the deletion test inline, like this:
```go
require.NotZero(t, out.ARN)
out, err := c.DeleteSecret(<stuff>)
require.NoError(t, err)
require.NotZero(t, out)
```

`defer cleanupSecret(t, out)` is only useful as an extra cleanup mechanism for tests that aren't explicitly already attempting to delete the secret immediately after creation. The only reason we're including the defer is to do extra cleanup to avoid littering AWS Secrets Manager with test secrets.